static/js/containers/pages/DashboardPage.js

```diff
@@ -173,6 +192,59 @@ export class DashboardPage extends React.Component<
     }
   }
 
+  renderEmailSettingsDialog(enrollment: RunEnrollment) {
+    const { emailSettingsModalVisibility } = this.state
+    return (
+      <Modal
+        className="text-center"
+        isOpen={emailSettingsModalVisibility}
+        toggle={this.toggleEmailSettingsModalVisibility}
+      >
+        <ModalHeader toggle={this.toggleEmailSettingsModalVisibility}>
+          Email Settings for {enrollment.run.course_number}
+        </ModalHeader>
+        <ModalBody>
+          <Formik
+            onSubmit={this.onSubmit.bind(this)}
+            initialValues={{
+              subscribeEmails: enrollment.edx_emails_subscription,
+              enrollmentId:    enrollment.id,
+              courseNumber:    enrollment.run.course_number
```

Comment:
We should add `course_number` in [courseTypes.js](https://github.com/mitodl/mitxonline/blob/main/static/js/flow/courseTypes.js#L29-L32)